### PR TITLE
`Offline Entitlements`: created `ProductEntitlementMappingResponse`

### PR DESCRIPTION
**Pull request by NachoSoto**

### CODE REVIEW

File: Sources/Networking/Responses/ProductsEntitlementsResponse.swift [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+//
+//      https://opensource.org/licenses/MIT
+//
+//  ProductsEntitlementsResponse.swift
+//
+//  Created by Nacho Soto on 3/17/23.
+
+import Foundation
+
+struct ProductsEntitlementsResponse {
```
Comment (NachoSoto) [inline Sources/Networking/Responses/ProductsEntitlementsResponse.swift:16]: I don't love this name, but changing it is easy, just wanted to get this blueprint structure out.

File: Tests/UnitTests/Networking/Responses/Fixtures/ProductsEntitlements.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+    "products": [
+        {
+            "id": "com.revenuecat.foo_1",
```
Comment (aboedo) [inline Tests/UnitTests/Networking/Responses/Fixtures/ProductsEntitlements.json:4]: maybe we should also change the endpoint response to be "identifier"? wdyt? I imagine with gzip compression enabled it wouldn't really make a difference in terms of data transfer

**Review by aboedo — APPROVED** [review]

**Review by tonidero — APPROVED** [review]

File: Tests/UnitTests/Networking/Responses/Fixtures/ProductsEntitlements.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+    "products": [
```
Comment (tonidero) [inline Tests/UnitTests/Networking/Responses/Fixtures/ProductsEntitlements.json:2]: Should this be `products_entitlements_mapping` or something like that?

File: Tests/UnitTests/Networking/Responses/Fixtures/ProductsEntitlements.json (hunk 2 above)
Comment (NachoSoto) [reply Tests/UnitTests/Networking/Responses/Fixtures/ProductsEntitlements.json:4]: I agree. Is this endpoint ready? This is just a placeholder until we have the real format from the backend.

File: Tests/UnitTests/Networking/Responses/Fixtures/ProductsEntitlements.json (hunk 3 above)
Comment (NachoSoto) [reply Tests/UnitTests/Networking/Responses/Fixtures/ProductsEntitlements.json:2]: I based this off of the suggested format from the doc.

**Comment by NachoSoto** [comment]
Merging this since it matches the agreed response format for now, but we can change it.

**Pull request merged**